protocol/protos/WifiCore.proto

```diff
@@ -131,3 +131,31 @@ enum AccessPointState
 {
     AccessPointStateUnknown = 0;
 }
+
+message WifiAuthenticationMethodConfigurationNone
+{
+
+}
+
+message WifiAuthenticationMethodConfigurationSharedKey
+{
+    Microsoft.Net.Wifi.SharedKey SharedKey = 1;
+}
+
+message WifiAuthenticationMethodConfigurationEnterprise
+{
+    string RadiusServerAddress = 1;
+    string RadiusServerPort = 2;
+    string RadiusServerSharedSecret = 3;
+}
+
+message WifiAuthenticationMethodConfiguration
+{
+    Microsoft.Net.Wifi.Dot11AuthenticationAlgorithm AuthenticationAlgorithm = 1;
+    oneof Configuration
+    {
+        WifiAuthenticationMethodConfigurationNone ConfigurationNone = 2;
+        WifiAuthenticationMethodConfigurationSharedKey ConfigurationSharedKey = 3;
+        WifiAuthenticationMethodConfigurationEnterprise ConfigurationEnterprise = 4;
+    }

```

Comment:
If I'm not mistaken, these `Configuration` values should cover everything needed for all `Dot11AuthenticationAlgorithm` values:

`DOT11_AUTH_ALGO_80211_OPEN`: `ConfigurationNone`
`DOT11_AUTH_ALGO_80211_SHARED_KEY`: `ConfigurationSharedKey`
`DOT11_AUTH_ALGO_WPA`: `ConfigurationEnterprise`
`DOT11_AUTH_ALGO_WPA_PSK`: `ConfigurationSharedKey`
`DOT11_AUTH_ALGO_WPA_NONE`: This is not supported
`DOT11_AUTH_ALGO_RSNA`: `ConfigurationEnterprise`
`DOT11_AUTH_ALGO_RSNA_PSK`: `ConfigurationSharedKey`
`DOT11_AUTH_ALGO_WPA3`: `ConfigurationEnterprise`
`DOT11_AUTH_ALGO_WPA3_ENT_192`: `ConfigurationEnterprise`
`DOT11_AUTH_ALGO_WPA3_SAE`: `ConfigurationSharedKey`
`DOT11_AUTH_ALGO_OWE`: `ConfigurationNone`
`DOT11_AUTH_ALGO_WPA3_ENT`: `ConfigurationEnterprise`